arangosh/Import/quick_hist.h

```diff
@@ -0,0 +1,265 @@
+        auto t = std::time(nullptr);
+        auto tm = *std::localtime(&t);
+
+        // new age string buffering & formatting
+        std::ostringstream oss;
+        oss << std::put_time(&tm, "%m-%d-%Y %H:%M:%S");
+        auto str = oss.str();
+
+        // old age string buffering & formatting
+        char buffer[133];
+        snprintf(buffer, sizeof(buffer), "%.3f,%.3f,%zd,%ld,%ld,%ld,%ld,%ld,%ld,%ld,%d,%s",
+               fp_measuring, fp_interval, num,
+               (0!=num) ? _readingLatencies->at(0).count() : 0,
+               mean.count(),median.count(),
+               per95.count(), per99.count(), per99_9.count(),
+               (0!=num) ? _readingLatencies->at(num-1).count() : 0,
+               0, str.c_str());
+        LOG_TOPIC(INFO, arangodb::Logger::FIXME)
+          << buffer;
+
+        _readingLatencies->clear();
+        _interval_start=interval_end;
+      }
+    }
+  }
+
+  //
+  // calculation taken from http://www.dummies.com/education/math/statistics/how-to-calculate-percentiles-in-statistics
+  //  (zero and one size vector calculations not included in that link)
+  std::chrono::microseconds CalcPercentile(std::vector<std::chrono::microseconds> &SortedLatencies, int Percentile) {
+    size_t index, remainder, next_index;
+    std::chrono::microseconds ret_val;
+
+    ret_val=std::chrono::microseconds(0);
+
+    if (0 < SortedLatencies.size()) {
+      // useful vector size
+
+      // percentiles are x10 ... so 95% is 950 and 99.9% is 999
+      index = SortedLatencies.size() * Percentile;
+      remainder = index % 1000;
+      index /= 1000;
+
+      // index is supposed to be x'th entry in list. x'th entry is 1 based, vector is 0 based,
+      //  then fix range
+      if (0 < index) {
+        --index;
+      } // if
+      next_index = ((index+1)<SortedLatencies.size()) ? index+1 : index;
+
+      if (0 == remainder) {
+        // whole number index
+        std::chrono::microseconds low, high;
+        low = SortedLatencies[index];
+        high = SortedLatencies[next_index];
+        ret_val = (low + high) / 2;
+      } else {
+        // fractional index, round up ... but we are 0 based, so already one higher
+        ret_val = SortedLatencies[next_index];
+      } // else
+    } else {
+      // vector too small
+      if (1 == SortedLatencies.size()) {
+        ret_val = SortedLatencies[0] / 2;
+      } else {
+        ret_val=std::chrono::microseconds(0);
+      } // else
+    } // else
+
+    return ret_val;
+  } // CalcPercentile
+
+ private:
+};
+
+
+class QuickHistogramTimer {
+public:
+  QuickHistogramTimer(QuickHistogram & histo)
+    : _histogram(histo) {
+    _interval_start=std::chrono::steady_clock::now();
+  }
+
+  ~QuickHistogramTimer() {
+    std::chrono::microseconds latency;
+
+    latency = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now() - _interval_start);
+    _histogram.post_latency(latency);
```

Comment:
may throw (calls push_back) and is not safe to call in a dtor.

Reply:
you cool with a try around the call and log on catch?